### ISSUE TITLE
Command is failing in version v0.10.0

### ISSUE THREAD

**Issue by karampok** [markdownlint/markdownlint#336]
## Description
Hi, there is probably an issue with the new version.
At least in our use case, it fails after the upgrade to version v0.10.0.

### Environment
**MDL Version**
v0.10.0

### Expected Behavior

```
▶ cat tools/md/Dockerfile
FROM alpine:3.10

RUN apk update && \
    apk add --no-cache ruby ruby-json && \
    gem install --no-rdoc --no-ri mdl -v 0.9.0

RUN mkdir /data

WORKDIR /data

ENTRYPOINT ["mdl"]
CMD ["--help"]

 ▶ docker build -q tools/md
sha256:03fead3154912113e52d7e2209c5679313241b54e34c4bf0e0f7d00348b42f5d
 ▶ docker run 03fead315491
Usage: mdl [options] [FILE.md|DIR ...]
    -c, --config FILE                The configuration file to use
    -g, --git-recurse                Only process files known to git when given a directory
    -i, --[no-]ignore-front-matter   Ignore YAML front matter
    -j, --json                       JSON output
    -l, --list-rules                 Don't process any files, just list enabled rules
```

### Actual Behavior

````
▶ cat tools/md/Dockerfile
FROM alpine:3.10

RUN apk update && \
    apk add --no-cache ruby ruby-json && \
    gem install --no-rdoc --no-ri mdl -v 0.10.0

RUN mkdir /data

WORKDIR /data

ENTRYPOINT ["mdl"]
CMD ["--help"]

 ▶ docker build -q tools/md
sha256:a146111c596fb562362d3a523d92805439e000f63f86d90f146587eb34548bf4

 ▶ docker run a146111c596fb562362d3a523d92805439e000f63f86d90f146587eb34548bf4
/usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require': cannot load such file -- etc (LoadError)
        from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
        from /usr/lib/ruby/gems/2.5.0/gems/mixlib-shellout-3.1.2/lib/mixlib/shellout.rb:19:in `<top (required)>'
        from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
        from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
        from /usr/lib/ruby/gems/2.5.0/gems/mdl-0.10.0/lib/mdl.rb:10:in `<top (required)>'
        from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
        from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
        from /usr/lib/ruby/gems/2.5.0/gems/mdl-0.10.0/bin/mdl:7:in `rescue in <top (required)>'
        from /usr/lib/ruby/gems/2.5.0/gems/mdl-0.10.0/bin/mdl:2:in `<top (required)>'
        from /usr/bin/mdl:23:in `load'
        from /usr/bin/mdl:23:in `<main>'
````

**Comment by jaymzh** [comment]
Can you reproduce this outside of docker? I cannot repro this locally:

```
[phil@rider (master) RVM(2.5.1) markdownlint]$ ./bin/mdl  --version
0.10.0
[phil@rider (master) RVM(2.5.1) markdownlint]$ ./bin/mdl  --help
Usage: mdl [options] [FILE.md|DIR ...]
    -c, --config FILE                The configuration file to use
    -g, --git-recurse                Only process files known to git when given a directory
    -i, --[no-]ignore-front-matter   Ignore YAML front matter
    -j, --json                       JSON output
    -l, --list-rules                 Don't process any files, just list enabled rules
    -r, --rules RULE1,RULE2          Only process these rules
    -u, --rulesets RULESET1,RULESET2 Specify additional ruleset files to load
    -a, --[no-]show-aliases          Show rule alias instead of rule ID when viewing rules
    -w, --[no-]warnings              Show kramdown warnings
    -d, --skip-default-ruleset       Don't load the default markdownlint ruleset
    -s, --style STYLE                Load the given style
    -t, --tags TAG1,TAG2             Only process rules with these tags
    -v, --[no-]verbose               Increase verbosity
    -h, --help                       Show this message
    -V, --version                    Show version
[phil@rider (master) RVM(2.5.1) markdownlint]$ 
```

**Comment by jaymzh** [comment]
Also can you show me the version of ruby and a full gem list in both containers?

**Comment by paul-at-cybr** [comment]
I'm experiencing the same issue (or an adjacent one), on the latest `ruby:alpine` and `circleci/ruby` images.
Reverting to 0.9.0 fixes the issue.

Output in our case:

```bash
/ # mdl -g .
Traceback (most recent call last):
	7: from /usr/local/bundle/bin/mdl:23:in `<main>'
	6: from /usr/local/bundle/bin/mdl:23:in `load'
	5: from /usr/local/bundle/gems/mdl-0.10.0/bin/mdl:10:in `<top (required)>'
	4: from /usr/local/bundle/gems/mdl-0.10.0/lib/mdl.rb:80:in `run'
	3: from /usr/local/bundle/gems/mdl-0.10.0/lib/mdl.rb:80:in `each'
	2: from /usr/local/bundle/gems/mdl-0.10.0/lib/mdl.rb:82:in `block in run'
	1: from /usr/local/bundle/gems/mdl-0.10.0/lib/mdl/doc.rb:50:in `new_from_file'
/usr/local/bundle/gems/mdl-0.10.0/lib/mdl/doc.rb:50:in `read': No such file or directory @ rb_sysopen - parrots/norwegianblue.md (Errno::ENOENT)
```

```bash
/ # ruby --version
ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux-musl]
```

```bash
/ # gem list

*** LOCAL GEMS ***

benchmark (default: 0.1.0)
bigdecimal (default: 2.0.0)
bundler (default: 2.1.4)
cgi (default: 0.1.0)
csv (default: 3.1.2)
date (default: 3.0.0)
dbm (default: 1.1.0)
delegate (default: 0.1.0)
did_you_mean (default: 1.4.0)
etc (default: 1.1.0)
fcntl (default: 1.0.0)
fiddle (default: 1.0.0)
fileutils (default: 1.4.1)
forwardable (default: 1.3.1)
gdbm (default: 2.1.0)
getoptlong (default: 0.1.0)
io-console (default: 0.5.6)
ipaddr (default: 1.2.2)
irb (default: 1.2.3)
json (default: 2.3.0)
kramdown (2.3.0)
kramdown-parser-gfm (1.1.0)
logger (default: 1.4.2)
matrix (default: 0.2.0)
mdl (0.9.0)
minitest (5.13.0)
mixlib-cli (2.1.6)
mixlib-config (3.0.6)
mutex_m (default: 0.1.0)
net-pop (default: 0.1.0)
net-smtp (default: 0.1.0)
net-telnet (0.2.0)
observer (default: 0.1.0)
open3 (default: 0.1.0)
openssl (default: 2.1.2)
ostruct (default: 0.2.0)
power_assert (1.1.7)
prime (default: 0.1.1)
pstore (default: 0.1.0)
psych (default: 3.1.0)
racc (default: 1.4.16)
rake (13.0.1)
rdoc (default: 6.2.1)
readline (default: 0.0.2)
readline-ext (default: 0.1.0)
reline (default: 0.1.3)
rexml (default: 3.2.3)
rss (default: 0.2.8)
sdbm (default: 1.0.0)
singleton (default: 0.1.0)
stringio (default: 0.1.0)
strscan (default: 1.0.3)
test-unit (3.3.4)
timeout (default: 0.1.0)
tomlrb (1.3.0)
tracer (default: 0.1.0)
uri (default: 0.10.0)
webrick (default: 1.6.0)
xmlrpc (0.3.0)
yaml (default: 0.1.0)
zlib (default: 1.1.0)
```

**Comment by jaymzh** [comment]
I can repro this in a docker build... but not locally. Not sure what's going on, but it seems like something about the alpine environment that's different. I'll try to find some time to debug tonight or tomorrow.

**Comment by brianeclow** [comment]
Dog piling on, I am unable to use the `--git-recurse` flag.

**Comment by dshevtsov** [comment]
I'm experiencing a similar issue on GitHub Action: https://github.com/magento/merchdocs/pull/654/checks?check_run_id=967045932#step:5:4

**Comment by jaymzh** [comment]
0.10.0 pushed to dockerhub. This should fix it.
Code didn't change so I'm not releasing a version just the Dockerfile to match

**Comment by ben-alkov** [comment]
Not a fix for Fedora/Centos.

This definitely broke with v0.10.0 available via `gem update mdl`.

It's broken in [Travis](https://travis-ci.org/github/containerbuildsystem/osbs-client/jobs/716909367) (Centos 7)/[GH action](https://github.com/containerbuildsystem/atomic-reactor/runs/971406132?check_suite_focus=true) (Fed 32)/locally (Fed 32)

N.B. This *only* occurs when using `-g`

(Context below is that I'm running locally in the root of the '[atomic-reactor](https://github.com/containerbuildsystem/atomic-reactor)' git repo.

```bash
10:02 🐚 mdl -g .
Traceback (most recent call last):
	7: from /home/USER/bin/mdl:23:in `<main>'
	6: from /home/USER/bin/mdl:23:in `load'
	5: from /home/USER/.gem/ruby/gems/mdl-0.10.0/bin/mdl:10:in `<top (required)>'
	4: from /home/USER/.gem/ruby/gems/mdl-0.10.0/lib/mdl.rb:80:in `run'
	3: from /home/USER/.gem/ruby/gems/mdl-0.10.0/lib/mdl.rb:80:in `each'
	2: from /home/USER/.gem/ruby/gems/mdl-0.10.0/lib/mdl.rb:82:in `block in run'
	1: from /home/USER/.gem/ruby/gems/mdl-0.10.0/lib/mdl/doc.rb:50:in `new_from_file'
/home/USER/.gem/ruby/gems/mdl-0.10.0/lib/mdl/doc.rb:50:in `read': No such file or directory @ rb_sysopen - CONTRIBUTING.md (Errno::ENOENT)
```

**Comment by jaymzh** [comment]
@ben-alkov - that's a different bug than the original reported bug here which is trying to load `etc` module. Please file a separate bug.

**Comment by zaventh** [comment]
This appears to still be an issue in Alpine and v0.11.0

```
FROM python:3-alpine
...
RUN apk add --no-cache bash build-base ruby ruby-dev \
    && gem install --no-document mdl:0.11.0 \
...
```

```
bash-5.0# mdl --help
Traceback (most recent call last):
        10: from /usr/bin/mdl:23:in `<main>'
         9: from /usr/bin/mdl:23:in `load'
         8: from /usr/lib/ruby/gems/2.7.0/gems/mdl-0.11.0/bin/mdl:3:in `<top (required)>'
         7: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
         6: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
         5: from /usr/lib/ruby/gems/2.7.0/gems/mdl-0.11.0/lib/mdl.rb:10:in `<top (required)>'
         4: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
         3: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
         2: from /usr/lib/ruby/gems/2.7.0/gems/mixlib-shellout-3.1.7/lib/mixlib/shellout.rb:19:in `<top (required)>'
         1: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
/usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require': cannot load such file -- etc (LoadError)
        11: from /usr/bin/mdl:23:in `<main>'
        10: from /usr/bin/mdl:23:in `load'
         9: from /usr/lib/ruby/gems/2.7.0/gems/mdl-0.11.0/bin/mdl:2:in `<top (required)>'
         8: from /usr/lib/ruby/gems/2.7.0/gems/mdl-0.11.0/bin/mdl:7:in `rescue in <top (required)>'
         7: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
         6: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
         5: from /usr/lib/ruby/gems/2.7.0/gems/mdl-0.11.0/lib/mdl.rb:10:in `<top (required)>'
         4: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
         3: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
         2: from /usr/lib/ruby/gems/2.7.0/gems/mixlib-shellout-3.1.7/lib/mixlib/shellout.rb:19:in `<top (required)>'
         1: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
/usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require': cannot load such file -- etc (LoadError)
```

**Comment by newmocart** [comment]
Just add ruby-etc package to alpine linux

```
apk add ruby-etc
```

that's all

**Comment by jaymzh** [comment]
@zaventh - you're using MDL 0.11.0, but that's not using the latest docker image we published when we released 0.11.0 which has ruby-etc in it.
